scripts/core/PROG.rst

```diff
@@ -1714,6 +1723,39 @@ The following table documents the supported knobs for overriding default functio
 | Memory          | ${X}_SHARED_FORCE_DEVICE_ALLOC        | {**0**, 1} | Forces all shared allocations into device memory                                  |
 +-----------------+-------------------------------------+------------+-----------------------------------------------------------------------------------+
 
+%endif
+
+%if ver >= 1.7:
++-----------------+-------------------------------------+-----------------+-----------------------------------------------------------------------------------+
+| Category        | Name                                | Values          | Description                                                                       |
++=================+=====================================+=================+===================================================================================+
+| Device          | ${X}_FLAT_DEVICE_HIERARCHY          | {**0**, 1, 2}   | Defines device hierarchy model exposed by Level Zero driver implementation        |
++                 +-------------------------------------+-----------------+-----------------------------------------------------------------------------------+
+|                 | ${X}_AFFINITY_MASK                  | list            | Forces driver to only report devices (and sub-devices) as specified by values     |
++                 +-------------------------------------+-----------------+-----------------------------------------------------------------------------------+
+|                 | ${X}_ENABLE_PCI_ID_DEVICE_ORDER     | {**0**, 1}      | Forces driver to report devices from lowest to highest PCI bus ID                 |
++-----------------+-------------------------------------+-----------------+-----------------------------------------------------------------------------------+
+| Memory          | ${X}_SHARED_FORCE_DEVICE_ALLOC      | {**0**, 1}      | Forces all shared allocations into device memory                                  |
++-----------------+-------------------------------------+-----------------+-----------------------------------------------------------------------------------+
+%endif
+
+%if ver >= 1.7:
+
+Device Hierarchy
+~~~~~~~~~~~~~
+
+${X}_FLAT_DEVICE_HIERARCHY allows users to select the device hierarchy model with which the underlying hardware is exposed and the types of devices returned with ${x}DeviceGet.
+
+With a value of `0`, ${x}DeviceGet returns all the devices that do not have a root-device. Traversing the device hierarchy is possible by querying sub-devices with ${x}DeviceGetSubDevices and root-devices with ${x}DeviceGetRootDevice. Driver implementation may perform implicit optimizations to submissions and allocations done in the root-devices.
```

Comment:
Recommend named constants for the values [0, 1, 2].

Reply:
@wdamon-intel : you mean like define enums? The main thing about enums is that we would have to give names to each enum, and so we would have to come up with names generic enough to be able to be used by any implementation. IMPLICIT_SCALING shouldn't be used, as that is Intel specific. do you have some names in mind?

Reply:
@wdamon-intel, @MichalMrozek  : what about:

0 -> COMPOSITE
1->  FLAT
2->  COMBINED or EXPANDEDFLAT or FLATWITHROOT (my preference here would be COMBINED).

Reply:
Added COMPOSITE, FLAT, COMBINED